src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -503,6 +511,19 @@ struct layout {
 
     ov::PartialShape transform(cldnn::format new_fmt) const;
 
+    size_t hash() const {
+        size_t seed = 0;
+        seed = hash_combine(seed, data_padding.hash());
+        seed = hash_combine(seed, format.value);
+        seed = hash_combine(seed, data_type);
+        if (!is_dynamic()) {
+            for (auto& d : get_shape()) {
+                seed = hash_combine(seed, d);
+            }
```

Comment:
Can't  we calculate the dynamic shape layout? I think we can use the partial shape value -1. 
Even though it is dynamic shape, it may have static dimensions partially. (e.g., {1, 2, -1, -1} .. I such a case, we might distinguish. (For now we only use this hash mechanism for impl cache with static shapes, but in the future we might need to extend the usage for compilation stage too..)

Reply:
I changed codes to get hash value using parital_shape. thanks.